pkg/host/internal/infiniband/ib_guid_config.go

```diff
@@ -0,0 +1,119 @@
+package infiniband
+
+import (
+	"encoding/json"
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/internal/lib/netlink"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/host/types"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/vars"
+)
+
+type ibPfGUIDJSONConfig struct {
+	PciAddress string   `json:"pciAddress,omitempty"`
+	PfGUID     string   `json:"pfGuid,omitempty"`
+	GUIDs      []string `json:"guids,omitempty"`
+	RangeStart string   `json:"rangeStart,omitempty"`
+	RangeEnd   string   `json:"rangeEnd,omitempty"`
+}
+
+type ibPfGUIDConfig struct {
+	GUIDs      []net.HardwareAddr
+	RangeStart GUID
+	RangeEnd   GUID
+}
+
+func getIbGUIDConfig(configPath string, netlinkLib netlink.NetlinkLib, networkHelper types.NetworkInterface) (map[string]ibPfGUIDConfig, error) {
+	links, err := netlinkLib.LinkList()
+	if err != nil {
+		return nil, err
+	}
+	rawConfigs, err := readJSONConfig(configPath)
+	if err != nil {
+		return nil, err
+	}
+	resultConfigs := map[string]ibPfGUIDConfig{}
+	// Parse JSON config into an internal struct
+	for _, rawConfig := range rawConfigs {
+		pciAddress, err := getPfPciAddressFromRawConfig(rawConfig, links, networkHelper)
+		if err != nil {
+			return nil, fmt.Errorf("failed to extract pci address from ib guid config: %w", err)
+		}
+		// GUID list and GUID range can't be set at the same time
```

Comment:
the comment here is not informative.

the immediate check below it does something else...

either comment for each check what is the intention or remov IMO.